### PR TITLE
[e2e] part 2 refactoring

### PR DESCRIPTION
**Pull request by pducolin**
Refactoring previous PR that added e2e tests for the installer script.

In this iteration no feature is added, only refactoring code to move common code.

Best reviewed per commit to follow the refactoring.

Next iteration will add one file per mode, and will add two new modes: fips and maximal install, with different assertions - you can see the next iteration in https://github.com/DataDog/agent-linux-install-script/pull/98/files#diff-89059d4760ebc9a893c31db5ec0d36a1305d6e1d82ad43b4d8331e97b8e6552b

### CODE REVIEW

**Comment by pducolin** [comment]
For reviewers: job succeded on the CI: https://gitlab.ddbuild.io/DataDog/agent-linux-install-script/-/jobs/351058245

File: test/e2e/agentflavor.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	}
+	*af = agentFlavor(value)
+	return nil
+}
+
+const (
+	agentFlavorDatadogAgent     agentFlavor = "datadog-agent"
+	agentFlavorDatadogIOTAgent  agentFlavor = "datadog-iot-agent"
+	agentFlavorDatadogDogstatsd agentFlavor = "datadog-dogstatsd"
+)
```
Comment (spencergilbert) [inline test/e2e/agentflavor.go:27]: **❓ question:** ‏ Do we know what additional changes be needed to support `obersvability-pipelines-worker` or `vector`? It seems out of scope for these changes, but would be good to understand what future work may entail.

**Review by spencergilbert — APPROVED** [review]

File: test/e2e/agentflavor.go (hunk 1 above)
Comment (pducolin) [reply test/e2e/agentflavor.go:27]: This test can potentially support any value supported by `DD_AGENT_FLAVOR`.

**Pull request merged**